var/spack/repos/builtin/packages/mpibind/package.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+from spack import *
+
+
+class Mpibind(AutotoolsPackage):
+    """A memory-driven algorithm to map parallel codes
+    to heterogeneous architectures"""
+
+    homepage    = "https://github.com/LLNL/mpibind"
+    url         = "https://github.com/LLNL/mpibind/archive/refs/tags/v0.5.0.tar.gz"
+    git         = "https://github.com/LLNL/mpibind.git"
+
+    maintainers = ['eleon']
+
+    # The build process uses 'git describe --tags' to get the
+    # package version, thus we need 'get_full_repo'
+    version('master', branch='master', get_full_repo=True)
+    version('0.5.0', sha256='51bb27341109aeef121a8630bd56f5551c70ebfd337a459fb70ef9015d97d2b7')
+
+    variant('cuda', default=False,
+            description='Build w/support for NVIDIA GPUs.')
+    variant('rocm', default=False,
+            description='Build w/support for AMD GPUs.')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+    depends_on('pkgconf',  type='build')
+
+    depends_on('hwloc@2:+libxml2',
+               type=('build', 'link'))
+    depends_on('hwloc@2:+pci', when=(sys.platform != 'darwin'),
+               type=('build', 'link'))
+    depends_on('hwloc@2:+cuda+nvml', when='+cuda',
+               type=('build', 'link'))
+    depends_on('hwloc@2.4:+rocm+opencl', when='+rocm',
+               type=('build', 'link'))
```

Comment:
I think you can simplify the recipe by removing the `type=` argument, since `(build, link)` is the default.

Reply:
Thanks for the suggestion. By explicitly having the type argument, however, the recipe is not at risk if the Spack defaults ever change.

Reply:
Spack defaults will likely change when we'll introduce (possibly after v0.17) separate concretization of build dependencies. Are you actually using any executable from `hwloc` as a tool you need to build this package? Otherwise the correct type would be just `link`.

Reply:
Good point, @alalazo. I believe you are correct, the `hwloc` dependency should be `link`. I will figure out how to change the PR to make these changes.